CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing to Heroic
+
+Thanks for taking the time to contribute! 
+
+The following is a set of guidelines for contributing to Heroic. 
+
+## Ground Rules
+
+* Be sure to add a comment to your PR explaining the reason for the change. Please include some context behind a bug or issue you ran in to, or your use case and why you need a particular feature added. 
+* Include tests for any large changes. Read here to learn how to run the [tests](https://github.com/spotify/heroic#testing).
+
+## Your First Contribution
+
+Unsure where to begin contributing to Heroic? You can start by browsing through these [starter issues](https://github.com/spotify/heroic/labels/level%3Astarter).
+
+Working on your first Pull Request? You can learn how from this free series, [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+
+## Getting started
+
+We use GitHub to manage issues. You can follow along and create issues [here](https://github.com/spotify/heroic/issues). 
+
+### Create your own fork of the code
+
+### Code style
+
+* Make sure you format the code using the provided formatter in [idea](https://github.com/spotify/heroic/blob/master/idea). Even if you disagree with the way it is formatted, consistency is more important. For special cases, see [Bypassing Validation](https://github.com/spotify/heroic#bypassing-validation).
+
+### Commit message conventions
+
+* If possible, limit your changes to one module per commit.
+If you add new, or modify existing classes. Keep that change to a single commit while maintaining
+backwards compatible behaviour. Deprecate any old APIs as appropriate with `@Deprecated` and
+add documentation for how to use the new API.
+
+* The first line of the commit should be formatted with `[module1,module2] my message`.
+
+* `module1` and `module2` are paths to the modules affected with any `heroic-` prefix stripped.
+So if your change affects `heroic-core` and `metric/bigtable`, the message should say
+`[core,metric/bigtable] did x to y`.
+
+* If more than _3 modules_ are affected by a commit, use `[all]`.
```

Comment:
We haven't been following this convention, but I like it. It would be interesting to see how many of our PRs end up touching multiple modules.